portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -142,4 +147,26 @@ public void doDelete(HttpServletRequest request, HttpServletResponse response) t
     DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
     datastore.delete(commentEntityKey);
   }
+
+  /**
+   * Method that returns parameter text's sentiment score
+   * Returns a value in [-1, 1], representing how negative or positive text is
+   */
+  private static float getSentimentScore(String text) throws IOException {
+      // Init doc element containing text
+      Document doc =
+        Document.newBuilder().setContent(text).setType(Document.Type.PLAIN_TEXT).build();
+
+      // Init the languageService
+      LanguageServiceClient languageService = LanguageServiceClient.create();
+    
+      // Analyze the doc's sentiment using the languageService and get a score
+      Sentiment sentiment = languageService.analyzeSentiment(doc).getDocumentSentiment();
+      float score = sentiment.getScore();
+
+      // Close the languageService because we no longer need it
+      languageService.close();
```

Comment:
Do you need to close it explicitly? What happens with the service when the variable goes out of scope?

Reply:
It closes automatically when it goes out of scope, so I don't need to close it. But AFAIK it's the standard for good practice.

Reply:
Well, let's agree to disagree :) To me, if the context gets closed (resource gets released) once a variable holding it goes out of scope, it is good practice to let the language do its thing. If you'd like to be explicit, that's okay, but the less code there is to read the better (to me).